oauth/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+use log::{debug, error, info, trace};
+use oauth2::reqwest::http_client;
+use oauth2::{
+    basic::BasicClient, AuthUrl, AuthorizationCode, ClientId, CsrfToken, PkceCodeChallenge,
+    RedirectUrl, Scope, TokenResponse, TokenUrl,
+};
+use std::io;
+use std::{
+    io::{BufRead, BufReader, Write},
+    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener},
+    process::exit,
+    sync::mpsc,
+};
+use url::Url;
+
+fn get_code(redirect_url: &str) -> AuthorizationCode {
+    let url = Url::parse(redirect_url).unwrap();
```

Comment:
Moving from 0.4 to 0.5 I spent a lot of time making the code panic-free by purging all `unwrap` and `expect`s. Could you have this and the below return `Result` too?

Reply:
OK, I worked through most of this last night. As a part-time Rust contributor, I'm guilty of unwrapping everything rather than handling errors properly so this stuff was a bit new. I probably haven't got it as nice as it could be but it's better.

Reply:
Looked through it. Unless I'm mistaken, I still see quite a lot of code that could panic due to unwrapping. How can I help you set it up to return e.g. a `Result<AuthorizationCode, core::Error>` (the other library parts should offer examples on how to but let me know if you need some pointers).

Reply:
I've worked on doing it but I've yet to push that work as I didn't quite finish. I'll try and sort it now.

Reply:
There we go